source/iostreams.tex

```diff
@@ -7566,32 +7689,149 @@
 
 \rSec3[stringbuf.members]{Member functions}
 
+\pnum
+The member functions getting the underlying character sequence
+all refer to a \tcode{high_mark} value,
+where \tcode{high_mark} represents the position
+one past the highest initialized character in the buffer.
+Characters can be initialized by writing to the stream,
+by constructing the \tcode{basic_stringbuf}
+passing a \tcode{basic_string} argument, or
+by calling one of the \tcode{str()} member functions
+passing a \tcode{basic_string} as an argument.
+In the latter case, all characters initialized prior to the call
+are now considered uninitialized
+(except for those characters re-initialized by the new \tcode{basic_string}).
+
+\begin{itemdecl}
+void init_buf_ptrs(); // \expos
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Initializes the input and output sequences from \tcode{buf}
+according to \tcode{mode}.
+
+\pnum
+\ensures
+\begin{itemize}
+\item If \tcode{ios_base::out} is set in \tcode{mode},
+  \tcode{pbase()} points to \tcode{buf.front()} and
+  \tcode{epptr() >= pbase() + buf.size()} is \tcode{true};
+  \begin{itemize}
+  \item in addition, if \tcode{ios_base::ate} is set in \tcode{mode},
+    \tcode{pptr() == pbase() + buf.size()} is \tcode{true},
+  \item otherwise \tcode{pptr() == pbase()} is \tcode{true}.
+  \end{itemize}
+\item If \tcode{ios_base::in} is set in \tcode{mode},
+  \tcode{eback()} points to \tcode{buf.front()}, and
+  \tcode{(gptr() == eback() \&\& egptr() == eback() + buf.size())}
+  is \tcode{true}.
+\end{itemize}
+
+\pnum
+\begin{note}
+For efficiency reasons
+stream buffer operations might violate invariants of \tcode{buf}
+while it is held encapsulated in the \tcode{basic_stringbuf},
+i.e., by writing to characters in the range
```

Comment:
i.e. -> e.g. (this is certainly not the only case where we could conceivably violate invariants of "buf")

Reply:
fixed